src/card/view/CardView.tsx

```diff
@@ -130,7 +131,18 @@ const NeoCardView = ({
     if (!settingsOpen) {
       setLastRunTimestamp(Date.now());
     }
-  }, [settingsOpen, query, JSON.stringify(localParameters)]);
+  }, [JSON.stringify(localParameters)]);
+
+  useEffect(() => {
+    if (!settingsOpen && selectorChange) {
+      setLastRunTimestamp(Date.now());
+    }
+    setSelectorChange(false);
+  }, [settingsOpen]);
+
+  useEffect(() => {
+    setSelectorChange(true);
```

Comment:
maybe we should explain here what `selectorChange` is.
I guess it means more something like "queryOrTypeChanged"?

Reply:
there's now an issue where queries don't refresh if you open/close the settings.
E.g. `return rand()`, do we want that intentionally?

Reply:
1. Comment added. 
2. We should encourage the use of the refresh button. We can set it on by default.